hmrc-mongo-play-26/src/main/scala/uk/gov/hmrc/mongo/play/json/Codecs.scala

```diff
@@ -75,57 +75,20 @@ trait Codecs {
   def toBson[A: Writes](a: A, legacyNumbers: Boolean = false): BsonValue =
     jsonToBson(legacyNumbers)(Json.toJson(a))
 
-  private def jsonToBson(legacyNumbers: Boolean)(js: JsValue): BsonValue =
-    js match {
-      case JsNull       => BsonNull.VALUE
-      case JsBoolean(b) => BsonBoolean.valueOf(b)
-      case JsNumber(n)  => if (legacyNumbers) toBsonNumberLegacy(n)
-                           else toBsonNumber(n)
-      case JsString(s)  => new BsonString(s)
-      case JsArray(a)   => new BsonArray(a.map(jsonToBson(legacyNumbers)).asJava)
-      case o: JsObject  => if (o.keys.exists(k => k.startsWith("$") && !List("$numberDecimal", "$numberLong").contains(k)))
-                             // mongo types, identified with $ in `MongoDB Extended JSON format`  (e.g. BsonObjectId, BsonDateTime)
-                             // should use default conversion to Json. Then PlayJsonReaders will then convert as appropriate
-                             // The exception are numbers handled above (otherwise precision of $numberDecimal will be lost)
-                             fromJsonDefault(o)
-                           else
-                             new BsonDocument(
-                               o.fields.map {
-                                 case (k, v) =>
-                                   new BsonElement(k, jsonToBson(legacyNumbers)(v))
-                               }.asJava
-                             )
-    }
-
-  // Following number conversion comes from https://github.com/ReactiveMongo/Play-ReactiveMongo/blob/4071a4fd580d7c6edeccac318d839456f69a847d/src/main/scala/play/modules/reactivemongo/Formatters.scala#L62-L64
-  // It will loose precision on BigDecimals which can't be represented as doubles, and incorrectly identify some large Doubles as Long.
-  // But is backward compatible with simple-reactivemongo
-  private def toBsonNumberLegacy(bd: BigDecimal): BsonValue =
-    if      (!bd.ulp.isWhole) new BsonDouble(bd.toDouble)
-    else if (bd.isValidInt)   new BsonInt32(bd.toInt)
-    else                      new BsonInt64(bd.toLong)
-
-  private def toBsonNumber(bd: BigDecimal): BsonValue =
-    if      (bd.isValidInt)      new BsonInt32(bd.intValue)
-    else if (bd.isValidLong)     new BsonInt64(bd.longValue)
-    else if (bd.isDecimalDouble) new BsonDouble(bd.doubleValue)
-    else                         // Not all bigDecimals are representable as Decimal128. Will throw [java.lang.NumberFormatException] with message: `Conversion to Decimal128 would require inexact rounding of -4.2176255923279509728936555398034786404E-54.`
-                                 new BsonDecimal128(new Decimal128(bd.bigDecimal))
-
-  private def bsonToJson(bs: BsonValue): JsValue =
+  def toJson(bs: BsonValue): JsValue =
```

Comment:
Does this need to be public? Client's should not be using the Json abstraction - They should be using `A` (or Bson) - the Json should be an internal implementation detail.
Can we keep the name `bsonToJson` to keep symmetry with the other internal `jsonToBson`. It is not the counterpart of `toBson` (which is a public `A -> Bson`)